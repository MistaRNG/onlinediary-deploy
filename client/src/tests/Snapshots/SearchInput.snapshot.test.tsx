import React from 'react';
import { render } from '@testing-library/react';
import SearchInput from '../../features/journal/components/Search/Input/';
import '@testing-library/jest-dom';
import { describe, it, expect } from '@jest/globals';
import { jest } from '@jest/globals';

describe('SearchInput Component Snapshot Tests', () => {
  it('should render the SearchInput component correctly with search text', () => {
    // ARRANGE
    const mockChangeHandler = jest.fn();
    const mockCancel = jest.fn();
    const mockSearch = 'Sample search';

    // ACT
    const { asFragment } = render(
      <SearchInput
        search={mockSearch}
        changeHandler={mockChangeHandler}
        cancel={mockCancel}
      />
    );

    // ASSERT
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the SearchInput component correctly without search text', () => {
    // ARRANGE
    const mockChangeHandler = jest.fn();
    const mockCancel = jest.fn();
    const mockSearch = '';

    // ACT
    const { asFragment } = render(
      <SearchInput
        search={mockSearch}
        changeHandler={mockChangeHandler}
        cancel={mockCancel}
      />
    );

    // ASSERT
    expect(asFragment()).toMatchSnapshot();
  });
});
