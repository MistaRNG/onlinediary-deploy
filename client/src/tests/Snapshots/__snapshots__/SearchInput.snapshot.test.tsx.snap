// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Component Snapshot Tests should render the SearchInput component correctly with search text 1`] = `
<DocumentFragment>
  <div
    class="bp4-input-group bp4-large search-input"
  >
    <input
      class="bp4-input"
      placeholder="Search..."
      style="padding-right: 0px;"
      type="text"
      value="Sample search"
    />
    <span
      class="bp4-input-action"
    >
      <button
        class="bp4-button bp4-minimal"
        type="button"
      >
        <span
          aria-hidden="true"
          class="bp4-icon bp4-icon-cross"
          icon="cross"
        >
          <svg
            data-icon="cross"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M9.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42L6.59 8 3.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`SearchInput Component Snapshot Tests should render the SearchInput component correctly without search text 1`] = `
<DocumentFragment>
  <div
    class="bp4-input-group bp4-large search-input"
  >
    <input
      class="bp4-input"
      placeholder="Search..."
      style="padding-right: 0px;"
      type="text"
      value=""
    />
    <span
      class="bp4-input-action"
    />
  </div>
</DocumentFragment>
`;
